tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/NetStandardLibraries/MsSql/EnterpriseLibraryClientExerciser.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#if NET462
+
+using System.Collections.Generic;
+using System.Data;
+using NewRelic.Agent.IntegrationTests.Shared;
+using NewRelic.Agent.IntegrationTests.Shared.ReflectionHelpers;
+using NewRelic.Api.Agent;
+using Microsoft.Practices.EnterpriseLibrary.Data.Sql;
+using sqlClient = System.Data.SqlClient;
+using System.Threading;
+
+namespace MultiFunctionApplicationHelpers.NetStandardLibraries.MsSql
+{
+    [Library]
+    public class EnterpriseLibraryClientExerciser : MsSqlExerciserBase
+    {
+        private static string _connectionString = MsSqlConfiguration.MsSqlConnectionString;
+
+        [LibraryMethod]
+        [Transaction]
+        public string MsSql(string tableName)
+        {
+            var teamMembers = new List<string>();
+            var msSqlDatabase = new SqlDatabase(_connectionString);
+
+            using (var reader = msSqlDatabase.ExecuteReader(CommandType.Text, "SELECT * FROM NewRelic.dbo.TeamMembers WHERE FirstName = 'John'"))
+            {
+                while (reader.Read())
+                {
+                    teamMembers.Add(reader.GetString(reader.GetOrdinal("FirstName")));
+                    if (reader.NextResult())
+                    {
+                        teamMembers.Add(reader.GetString(reader.GetOrdinal("FirstName")));
+                    }
+                }
+            }
+
+            var insertSql = string.Format(InsertPersonMsSql, tableName);
+            var countSql = string.Format(CountPersonMsSql, tableName);
+            var deleteSql = string.Format(DeletePersonMsSql, tableName);
+            _ = msSqlDatabase.ExecuteNonQuery(CommandType.Text, insertSql);
```

Comment:
Do we instrument any async methods that need to be exercised also?

Reply:
Nope, only sync for EnterpriseLibrary.